files/logs.py

```diff
@@ -16,21 +16,28 @@ def __init__(self, objectName):
 		self.objectName = objectName
 
 	def __print__(self, fields):
-		fields['level'] = self.level
-		fields['objectName'] = self.objectName
-		fields['date'] = datetime.now().strftime("%Y-%m-%d %H:%M:%S")
-
+		# These are the fields are printed all the time in each log line, in the following order.
+		mandatoryFields = {
+			'date': datetime.now().strftime("%Y-%m-%d %H:%M:%S"),
+			'level': self.level,
+			'objectName': self.objectName,
+			'ip': '',
+			'referrer': ''
+		}
 		# Try to get IP and referrer from user
 		try:
-			fields['ip'] = request.environ.get('HTTP_X_REAL_IP', request.remote_addr)
-			fields['referrer'] = request.headers.get("Referer")
+			mandatoryFields['ip'] = request.environ.get('HTTP_X_REAL_IP', request.remote_addr)
+			mandatoryFields['referrer'] = request.headers.get("Referer")
 		except Exception as e:
 			pass
 
+		# Merge dictionaries
+		mandatoryFields.update(fields)
```

Comment:
I would do something like 
`allFields = mandatoryFields.update(fields)`

otherwise when you keep reading the code bellow it seems you only print the mandatoryFields, you have to read above to realize that at that point mandatoryFields contains all the fields